### PR TITLE
#301 bit.ly links are broken

### PR DESCRIPTION
**Pull request by dmzaytsev**
For #301
- Current cache implementation works inside Doc instance, but we create docs on every page load. It makes cache useless. Implemented single static cache for short urls to avoid this.
- improved the unit test

### CODE REVIEW

**Comment by dmzaytsev** [comment]
@karato travis failure is unrelated to issue. see #299

**Comment by karato** [comment]
@dmzaytsev I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by karato** [comment]
@krzyk review it

File: src/main/java/com/nerodesk/om/aws/CdShortUrl.java [hunk 1]
```diff
@@ -50,6 +52,10 @@
 @EqualsAndHashCode
```
Comment (krzyk) [inline src/main/java/com/nerodesk/om/aws/CdShortUrl.java]: @dmzaytsev I think that the only thing you need for this to work is `@EqualsAndHashCode(of = 'decorated')`

**Comment by krzyk** [comment]
@dmzaytsev I think that the problem is because `@EqualsAndHashcode` generated (which is used in cache) will be the default (hashcode = 1, and equals will be always true for object of the same class). This annotation ignores `transient` fields so we have to add `of = {}` to it.
With that I think it should work without the added seperate cache.

**Comment by krzyk** [comment]
@dmzaytsev also you have to make sure that all the `Doc` implementers have correct equals and hashcode (from what I see `SmallDoc` has the same problem as your implementation, and MkDoc doesn't have any)

**Comment by dmzaytsev** [comment]
@krzyk I don't know well enough internal implementation jcabi-cache.
anyway I will follow to your advice

**Comment by dmzaytsev** [comment]
@krzyk fixed
test passed localy

**Comment by krzyk** [comment]
@dmzaytsev we will need to wait for #299 to merge

**Comment by krzyk** [comment]
@karato we are waiting for #299

**Comment by karato** [comment]
> @karato we are waiting for #299 

@krzyk yes, right, let's wait for #299

**Comment by krzyk** [comment]
@dmzaytsev one more comment, please add a `conformsToEqualsHashCodeContract` in both `SmallDocTest` and `CdShortUrlTest`, you can use the one from e.g. `AwsUserTest`. This way we will know sooner if some fields are not taken into account when doing hascode or equals.

**Comment by dmzaytsev** [comment]
@krzyk what about other Docs? e.q. SafeDoc

**Comment by krzyk** [comment]
@dmzaytsev I forgot there are others, yes we need to have it in all the docs.

**Comment by dmzaytsev** [comment]
@krzyk I added these tests

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by yegor256** [comment]
@rultor merge

**Comment by karato** [comment]
@krzyk **22 mins** was added to your account, many thanks for your contribution! The task took 58539002.... review comments (c=7) added as a bonus... +22 to your rating, your total score is [+4623](http://www.netbout.com/b/35035?open=rating)

**Comment by karato** [comment]
@rultor deploy

**Pull request merged**